src/ansi.rs

```diff
@@ -0,0 +1,1821 @@
+//! Process parsed terminal streams into terminal actions.
+
+use core::convert::TryFrom;
+use core::{iter, str};
+
+use core::time::Duration;
+
+#[cfg(feature = "alloc")]
+use alloc::string::String;
+#[cfg(feature = "alloc")]
+use alloc::vec::Vec;
+
+#[cfg(not(feature = "alloc"))]
+use arrayvec::ArrayVec;
+use log::{debug, trace};
+
+use crate::{Params, ParamsIter, Parser, Perform};
+use core::option::Option::Some;
+
+#[derive(Debug, Eq, PartialEq, Copy, Clone, Default)]
+pub struct Rgb {
+    pub r: u8,
+    pub g: u8,
+    pub b: u8,
+}
+
+/// Maximum time before a synchronized update is aborted.
+const SYNC_UPDATE_TIMEOUT: Duration = Duration::from_millis(150);
+
+/// Maximum number of bytes read in one synchronized update (2MiB).
+#[cfg(feature = "alloc")]
+const SYNC_BUFFER_SIZE: usize = 0x20_0000;
+
+/// Maximum number of bytes read in one synchronized update (4Kib) when no_alloc feature is set.
+#[cfg(not(feature = "alloc"))]
+const SYNC_BUFFER_SIZE_NO_ALLOC: usize = 0x4000;
```

Comment:
I'm afraid a buffer size of 4KiB would make this feature utterly useless.

Reply:
Is there any buffer size, which makes this feature work for most use cases? I don't think that this library should support _all_ use cases in no_std systems without allocator.

Reply:
Yeah, the existing code uses a buffer limit already. we're using 2MiB by default, I don't think it makes sense to cut this below 1 MiB.

So that's probably not going to fly.

Reply:
Ok, I disabled all `DCS` handling when `alloc` is not set

Reply:
@chrisduerr So what should I do?
- We cannot preallocate 1 MiB buffer, because it will inflate resulting binary and make this crate unsuitable for embedded systems
- We cannot use buffer smaller than 1 MiB
- We cannot disable `DCS` feature when no allocator is present

Reply:
My problem's just with disabling `DCS` handling in general. I have no problem with disabling the sync updates escape.

Reply:
Hi, sorry for the delay. Yes, I tried to disable sync updates only, but I can't see a clear boundary between `DCS` and sync updates handling, so I did that I could.